File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = []
    obj = {}
    
    for i in range(len(name)):
        obj[name[i]] = yearning[i]
    
    for items in photo:
        sumValue = 0
        for item in items:
            if item in obj:
                sumValue += obj[item]
        answer.append(sumValue)
        
    return answer

File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    const answer = [];
    const ref = {};
    name.forEach((item, index) => {
        ref[item] = yearning[index];
    })
    
    photo.forEach((item) => {
        let result = 0
        item.forEach((targetName) => {
            result += ref[targetName] || 0;
        })
        answer.push(result);
    })
    return answer;
}